src/txn/txn_recover.c

```diff
@@ -484,9 +484,10 @@ __recovery_set_checkpoint_snapshot(WT_SESSION_IMPL *session)
 
     /*
      * WiredTiger versions 10.0.1 onward have a valid checkpoint snapshot on-disk. Ignore reading
-     * the on-disk checkpoint snapshot from older versions.
+     * the on-disk checkpoint snapshot from older versions. WiredTiger backup doesn't contain the
+     * WiredTiger.turtle file. Allow retrieving the saved checkpoint snapshot if it exists.
      */
-    if (conn->recovery_major < 10 ||
+    if ((conn->recovery_major != 0 && conn->recovery_major < 10) ||
```

Comment:
I believe this change is right, but it makes me uncomfortable because it relies on the explicit 0 initialization of the WT_CONNECTION structure, and that not being overwritten in an unrelated `conn_version_verify` function.

Could you also update `conn_version_verify` to deal explicitly with the case of having no turtle file. That might just be a comment update, but it could also be an explicit setting of the versions to no-op values.

Reply:
Initialised the variables in the `conn_version_verify` function to avoid the dependency of connection structure being initialised to 0.

Reply:
Heh - and I just left a big comment about the addition of that (it shows earlier in the diff so I saw that before opening this resolved comment). @agorrod I find initializing a few fields in the conn structure to 0 more confusing when it is only called early in wiredtiger_open. That makes me have to dig around to wonder why it wouldn't be 0 from the calloc. There's a lot of code that depends on the connection being zeroed on allocation. We don't initialize every other field explicitly in random places.